### DISCUSSION TITLE
Error decrypting message: Session key decryption failed.

### DISCUSSION THREAD

**Discussion by varsha123** [openpgpjs/openpgpjs#1520] [Q&A] [answered]
Hi Team

we are facing below error : while decrypting signencrypted file generated using Bouncycastle library

Error decrypting message: Session key decryption failed.

Also same file we have tried with GPG tool: we received below detail: plz suggest what's wrong here in this file or our process

gpg: encrypted with rsa3072 key, ID ###############, created 2021-10-26
"Tazapay Devops (Tazapay SCB dev PGP) <#########>"
gpg: used key is not marked for encryption use.
File 'doc' exists. Overwrite? (y/N) n
Enter new filename: doc1
gpg: Signature made 12-May-22 11:31:45 AM India Standard Time
gpg: using RSA key ###################
gpg: issuer "##############"
gpg: Good signature from "###############" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg: There is no indication that the signature belongs to the owner.
Primary key fingerprint: #########################################
gpg: WARNING: message was not integrity protected
gpg: decryption forced to fail!

GPG version :
gpg (GnuPG) 2.3.6
libgcrypt 1.10.1
Copyright (C) 2021 g10 Code GmbH
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.

Please let us know how to fix this.

below is our decryption code

   const messagedecrypt:any =fs.readFileSync("TESTPGP_PGPOUT", "utf8");
    const message:any = await openpgp.readMessage({
        armoredMessage: messagedecrypt, // parse armored message       
    });
  
    const { data: decrypted, signatures } = await openpgp.decrypt({
        message,
        verificationKeys: publicKey, // optional
        decryptionKeys: privateKey1
    });
    console.log("decrypted-----");
    console.log(decrypted); // 'Hello, World!'
    console.log(message.packets[0].publicKeyId);

    try {
        await signatures[0].verified; // throws on invalid signature
        console.log('Signature is valid');
    } catch (e) {
        throw new Error('Signature could not be verified: ' + e.message);
    }

Thanks
Varsha

**Comment by larabr** [comment, marked as answer]
Hi @varsha123 , the message has multiple security-relevant issues, which is why also GPG refuses to decrypt it by default.
I recommend to upgrade/change encryption software since the encryption mechanism without integrity protection is being deprecated, and using signing keys for decryption is not safe with RSA keys.

To get around the security checks that OpenPGP.js performs, you need to pass different configuration options to `openpgp.decrypt`. Specifically, for :

> WARNING: message was not integrity protected

you can pass `config.allowUnauthenticatedMessages: true` -- find more details at https://github.com/openpgpjs/openpgpjs/discussions/1418 and https://github.com/openpgpjs/openpgpjs/blob/main/src/config/config.js#L85-L93.

While for:

> gpg: used key is not marked for encryption use.

use `config.allowInsecureDecryptionWithSigningKeys: true` , more info at https://github.com/openpgpjs/openpgpjs/discussions/1352#discussioncomment-931652 .

To apply the configuration, you can call
```js
openpgp.decrypt({
    message,
    verificationKeys: publicKey,
    decryptionKeys: privateKey1,
    config: { allowInsecureDecryptionWithSigningKeys: true, allowUnauthenticatedMessages: true }
});
```
